dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -179,23 +179,93 @@ def _add_helper_merge_profile_histograms(
+            ideal_count_of_bins = self.user_set_histogram_bin
+
         for method in self.histogram_bin_method_names:
             self.histogram_methods[method] = {
                 "total_loss": 0,
                 "current_loss": 0,
                 "histogram": {"bin_counts": None, "bin_edges": None},
             }
 
-        combined_values = np.concatenate(
-            [other1._histogram_to_array(), other2._histogram_to_array()]
+        # calculate the min of the first edge and the max of the last edge
+        # between two arrays
+        global_min_of_histogram_edges = (
+            float(self.min)
+            if self.min is not None
+            else min(
+                other1._stored_histogram["histogram"]["bin_edges"][0],
+                other2._stored_histogram["histogram"]["bin_edges"][0],
+            )
         )
-        bin_counts, bin_edges = self._get_histogram(combined_values)
-        self._stored_histogram["histogram"]["bin_counts"] = bin_counts
-        self._stored_histogram["histogram"]["bin_edges"] = bin_edges
 
-        histogram_loss = self._histogram_bin_error(combined_values)
-        self._stored_histogram["histogram"]["current_loss"] = histogram_loss
-        self._stored_histogram["histogram"]["total_loss"] = histogram_loss
+        global_max_of_histogram_edges = (
+            float(self.max)
+            if self.max is not None
+            else max(
+                other1._stored_histogram["histogram"]["bin_edges"][-1],
+                other2._stored_histogram["histogram"]["bin_edges"][-1],
+            )
+        )
+
+        # Generate new bin edges
+        ideal_bin_edges = np.linspace(
+            global_min_of_histogram_edges,
+            global_max_of_histogram_edges,
+            ideal_count_of_bins + 1,
+        )
+
+        # Initialize new count by bin object for population
+        new_entity_count_by_bin = np.zeros((ideal_count_of_bins,))
+
+        # Generate new histograms
+        _, histogram_loss_1 = self._assimilate_histogram(
+            from_hist_entity_count_per_bin=other1._stored_histogram["histogram"][
+                "bin_counts"
+            ],
+            from_hist_bin_edges=other1._stored_histogram["histogram"]["bin_edges"],
+            dest_hist_entity_count_per_bin=new_entity_count_by_bin,
+            dest_hist_bin_edges=ideal_bin_edges,
+            dest_hist_num_bin=ideal_count_of_bins,
+        )
+
+        # Ensure loss is calculated on second run of regenerate
+        _, histogram_loss_2 = self._assimilate_histogram(
+            from_hist_entity_count_per_bin=other2._stored_histogram["histogram"][
+                "bin_counts"
+            ],
+            from_hist_bin_edges=other2._stored_histogram["histogram"]["bin_edges"],
+            dest_hist_entity_count_per_bin=new_entity_count_by_bin,
+            dest_hist_bin_edges=ideal_bin_edges,
+            dest_hist_num_bin=ideal_count_of_bins,
+        )
+        try:
```

Comment:
we could separate out getting sizes from the actual calc so the calc is not within the try itself. that is if sizes are needed.

Reply:
see below